### PR TITLE
Fix PinotSegmentReatletResource APIs when raw table name is provided and table type is not specified

### PR DESCRIPTION
**Pull request by Jackie-Jiang**
Added a helper method getExistingTableNamesWithType to get all existing tables with the given table name and table type

### CODE REVIEW

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotSegmentRestletResource.java [hunk 1]
```diff
@@ -291,42 +281,73 @@ public String getCrcForAllSegments(
     return getAllCrcMetadataForTable(tableName);
   }
 
-  //////////////////////////////////////////////////////////////////////////////////////
+  private List<String> getExistingTableNamesWithType(@Nonnull String tableName,
```
Comment (snleee) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotSegmentRestletResource.java]: Can you add the documentation for this helper function?

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotSegmentRestletResource.java [hunk 2]
```diff
@@ -481,26 +487,23 @@ private JSONArray getAllSegmentsMetadataForTable(String tableName, CommonConstan
    * XXX
```
Comment (snleee) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotSegmentRestletResource.java]: I know that you didn't add this but I think we should remove this?

Comment (Jackie-Jiang) [reply pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotSegmentRestletResource.java]: Done

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotSegmentRestletResource.java (hunk 1 above)
Comment (Jackie-Jiang) [reply pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotSegmentRestletResource.java]: Done

**Pull request merged**